caproto/asyncio/client.py

```diff
@@ -1071,18 +1065,9 @@ async def send(self, *commands, extra=None):
         # Turn the crank: inform the VirtualCircuit that these commands will be
         # send, and convert them to buffers.
         buffers_to_send = self.circuit.send(*commands, extra=extra)
-
-        # lock to make sure a AddEvent does not write bytes to the socket while
-        # we are sending
-        async def _socket_send(buffers_to_send):
-            'Send a list of buffers over the socket'
-            try:
-                return self.transport.sock.sendmsg(buffers_to_send)
-            except BlockingIOError:
-                raise ca.SendAllRetry()
-
-        async with self._raw_lock:
-            await ca.async_send_all(buffers_to_send, _socket_send)
+        # TODO: can we trust asyncio to get this all out the door 😱
```

Comment:
Not sure we can. I seem to recall issues with large arrays back in the day.

This is a pretty major change though for the client, also requiring that the joined buffer be created rather than using the buffers by reference as created in `circuit.send`.

Reply:
fair.

After I wrote this I did a bit more reading and I think there is a better option that does not require the joining.